eventsdiscovery/src/main/java/com/schedjoules/client/eventsdiscovery/json/JsonEvent.java

```diff
@@ -64,10 +68,10 @@ else if (mJsonObject.has("timeZone"))
 
 
     @Override
-    public Duration duration()
+    public Optional<Duration> duration()
     {
-        String durationString = mJsonObject.optString("duration", "P0D");
-        return Duration.parse(durationString);
+        return mJsonObject.has(KEY_DURATION) && !mJsonObject.isNull(KEY_DURATION) ?
```

Comment:
`mJsonObject.has(KEY_DURATION)` could be removed here now, otherwise looks good.